libgalois/include/katana/PropertyGraph.h

```diff
@@ -359,16 +396,32 @@ class KATANA_EXPORT PropertyGraph {
 
   PropertyGraph() = default;
 
+  // PropertyGraph(
+  //     std::unique_ptr<tsuba::RDGFile>&& rdg_file, tsuba::RDG&& rdg,
+  //     GraphTopology&& topo) noexcept
+  //     : rdg_(std::move(rdg)),
+  //       file_(std::move(rdg_file)),
+  //       topology_(std::move(topo)) {}
+
+  /// TODO: Giorgi, add a new constructor factory method to include the new way of storing types.
   PropertyGraph(
       std::unique_ptr<tsuba::RDGFile>&& rdg_file, tsuba::RDG&& rdg,
-      GraphTopology&& topo) noexcept
+      GraphTopology&& topo,
+      katana::NUMAArray<TypeSetID>&& node_type_ids,
+      katana::NUMAArray<TypeSetID>&& edge_type_ids
+      ) noexcept
       : rdg_(std::move(rdg)),
         file_(std::move(rdg_file)),
-        topology_(std::move(topo)) {}
+        topology_(std::move(topo)),
+        node_type_set_id_(std::move(node_type_ids)),
```

Comment:
What about the maps, would those be added too?

Reply:
Yes

Reply:
Actually, the maps will be passed in through the `rdg_` .